packages/odf/modals/s3-iam/UpdateAccessKeyModal.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import { StatusType } from '@aws-sdk/client-iam';
+import { AccessKeyStatus } from '@odf/core/constants/s3-iam';
+import { useCustomTranslation } from '@odf/shared';
+import { ButtonBar } from '@odf/shared/generic/ButtonBar';
+import { IamCommands } from '@odf/shared/iam';
+import { CommonModalProps } from '@odf/shared/modals';
+import * as _ from 'lodash-es';
+import useSWRMutation from 'swr/mutation';
+import {
+  Button,
+  ButtonVariant,
+  Flex,
+  FlexItem,
+  Form,
+  FormGroup,
+  Modal,
+  ModalVariant,
+} from '@patternfly/react-core';
+
+export type UpdateAccessKeyModalProps = {
+  name: string;
+  AccessKeyId: string;
+  status: StatusType;
+  iamClient: IamCommands;
+};
+
+const UpdateAccessKeyModal: React.FC<
+  CommonModalProps<UpdateAccessKeyModalProps>
+> = ({
+  closeModal,
+  isOpen,
+  extraProps: { name, AccessKeyId, status, iamClient },
+}) => {
+  const { t } = useCustomTranslation();
+
+  const {
+    trigger: updateAccessKey,
+    isMutating: inProgress,
+    error,
+    reset,
+  } = useSWRMutation(`update-access-key-${name}-${AccessKeyId}`, async () => {
```

Comment:
```suggestion
  } = useSWRMutation(`update-access-key-${name}-${AccessKeyId}`, async () => {
```
Why do we need SWR mutation for updating key too?